src/Arcus.Security.Providers.AzureKeyVault/Extensions/SecretStoreBuilderExtensions.cs

```diff
@@ -263,7 +266,69 @@ public static class SecretStoreBuilderExtensions
+            return WithCachedSecretProvider(builder, keyVaultSecretProvider, cacheConfiguration, mutateSecretName);
+        }
+
+        /// <summary>
+        /// Adds Azure Key Vault as a secret source.
+        /// </summary>
+        /// <param name="builder">The builder to create the secret store.</param>
+        /// <param name="tokenCredential">The requested authentication type for connecting to the Azure Key Vault instance.</param>
+        /// <param name="configuration">The configuration related to the Azure Key Vault instance to use.</param>
+        /// <param name="mutateSecretName">The optional function to mutate the secret name before looking it up.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="builder"/>, <paramref name="tokenCredential"/>, or <paramref name="configuration"/> is <c>null</c>.</exception>
+        public static SecretStoreBuilder AddAzureKeyVault(
+            this SecretStoreBuilder builder,
+            TokenCredential tokenCredential,
+            IKeyVaultConfiguration configuration,
+            Func<string, string> mutateSecretName = null)
+        {
+            Guard.NotNull(builder, nameof(builder), "Requires a secret store builder to add the Azure Key Vault secret provider");
+            Guard.NotNull(tokenCredential, nameof(tokenCredential), "Requires an Azure Key Vault authentication instance to add the secret provider to the secret store");
+            Guard.NotNull(configuration, nameof(configuration), "Requires an Azure Key Vault configuration instance to add the secret provider to the secret store");
+
+            return AddAzureKeyVault(builder, tokenCredential, configuration, cacheConfiguration: null, mutateSecretName: mutateSecretName);
+        }
+
+        /// <summary>
+        /// Adds Azure Key Vault as a secret source.
+        /// </summary>
+        /// <param name="builder">The builder to create the secret store.</param>
+        /// <param name="tokenCredential">The requested authentication type for connecting to the Azure Key Vault instance.</param>
+        /// <param name="configuration">The configuration related to the Azure Key Vault instance to use.</param>
+        /// <param name="cacheConfiguration">The configuration to control how the caching will be done.</param>
+        /// <param name="mutateSecretName">The optional function to mutate the secret name before looking it up.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="builder"/>, <paramref name="tokenCredential"/>, or <paramref name="configuration"/> is <c>null</c>.</exception>
+        public static SecretStoreBuilder AddAzureKeyVault(
+            this SecretStoreBuilder builder,
+            TokenCredential tokenCredential,
+            IKeyVaultConfiguration configuration,
+            ICacheConfiguration cacheConfiguration,
+            Func<string, string> mutateSecretName = null)
```

Comment:
Added here a new extension with the base class `TokenCredential`; which is the base class for all the authentication types. (So service principal, managed identity, ...). Should we provide something simulair as to what we support or with the specific types (as in `ManagedIdentityCredential`), or something simulair like what we already have with the required values (ex. `clientId` and `clientSecret`) in the extension itself? We may have some conflicts there, though.

Reply:
Didn't add docs just yet because it all depends on what we support here.

Reply:
I personally would use both:
- We have the convenient extensions such as `AddAzureKeyVaultWithCertificate` which just take the required input. However, I would change the current ones to use Azure.Identity, not add new ones.
- We provide extensions to pass the TokenCredential directly in case they want to manage it.

Reply:
> Are you planning on writing integration tests for this as well?

Well, if we would change the existing ones, there's not so much additional integration tests to be written. Except the one where we pass along the `TokenCredential` ourselves.

Reply:
If we change the existing ones, we would have breaking changes because the `TokenCredential` implementations (such as `ClientCertificateCredential`) also takes in the `tenantId` which is now not currently available. Should I go through with the breaking change or should I mark them obsolete and create new ones.

Reply:
> Should I go through with the breaking change or should I mark them obsolete and create new ones.

Let's add new ones and deprecate the current ones